src/NuGet.Core/NuGet.Packaging/Signing/Native/NativeCmsWrapper .cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGet.Packaging.Signing
+{
+#if IS_SIGNING_SUPPORTED && IS_DESKTOP
+    internal class NativeCmsWrapper : ICms
+    {
+        private NativeCms _nativeCms;
+
+        public NativeCmsWrapper(NativeCms nativeCms)
```

Comment:
Thanks! According to CmsFactory, the nativeCms will only be null if the framework is not supported. Since I added throwing NotSupportedException in CmsFactory for unsupported framework, do I still need to do the null check? If so, what shall do if it's null? Thanks!

Reply:
This is a standalone class, so basic argument checking is a good idea.

However, how about adding argument checking to `CmsFactory` instead?

Reply:
Thanks! Added checking in `CmsFactory `.